application-numbered-content-headings/application-numbered-content-headings-ui/src/main/resources/NumberedHeadings/Code/ActivationUI.xml

```diff
@@ -239,6 +239,143 @@ require(['jquery', 'NumberedHeadingsConfig', 'bootstrap'], function($, config) {
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <restricted>0</restricted>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'deferred!ckeditor'], function($, ckeditorPromise) {
+  ckeditorPromise.done(ckeditor =&gt; {
+    if (!('xwiki-numberedContent-activationUI' in ckeditor.plugins.registered)) {
+      ckeditor.plugins.add('xwiki-numberedContent-activationUI', {
+        init: function(editor) {
+          // Refresh the content of the editor. The 'xwiki:wysiwyg:convertHTML' listened below will be triggered indirectly.
+          $('#NumberedHeadings\\.Code\\.NumberedHeadingsClass_0_status').on('change', function() {
+            editor.execCommand('xwiki-refresh');
+          });
+        }
+      });
+
+      // Read the current configuration of numbered heading in the select of the UIX and provide it for the convertHTML request.
+      $(document).on('xwiki:wysiwyg:convertHTML', function(event, data) {
+        const value = $('#NumberedHeadings\\.Code\\.NumberedHeadingsClass_0_status').val();
+        var enableNumberedHeadings = false;
+
+        if (value === '') {
+          var numberedHeadingsConfig = JSON.parse(document.getElementById('numbered-headings-config')
+            .textContent);
+          enableNumberedHeadings = numberedHeadingsConfig.isActivatedOnParent;
```

Comment:
Why isn't this using the `NumberedHeadingsConfig` module instead of parsing the JSON again?